client/src/CourseView/CourseView.jsx

```diff
@@ -1,24 +1,26 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import { ListGroup, ListGroupItem } from 'reactstrap';
 import CourseViewModal from "./CourseViewModal";
-
-const data = [
-    {name: 'Hulgateooria', code: 'IF9120',  description: "hulgateooria sissejuhatus"},
-    {name: 'Interaktsioonidisain', code: 'IF1230',  description: "interaktsioonidisaini sissejuhatus"},
-    {name: 'Infosüsteemid', code: 'IFI9535', description: "infosüsteemide sissejuhatus"}
-];
+import axios from 'axios';
 
 export const CourseView = () => {
-  const [isModalVisible, setModalVisible] = useState(false);
+  const [data, setData] = useState([]);
+    useEffect( () => {
+      async function fetchData() {
+        const response = await axios('http://localhost:8080/api/v1/courses');
```

Comment:
This needs error handling. Also, we have a proxy, the request should work => await axios ('/api/v1/courses'')